ooniprobe/Engine/Engine.m

```diff
@@ -1,40 +1,90 @@
 #import "Engine.h"
 #import <oonimkall/Oonimkall.h>
-#import <mkall/MKVersion.h>
-#define probeEngineTasks @[@"Telegram", @"Ndt", @"Dash", @"Psiphon", @"Tor", @"Whatsapp", @"FacebookMessenger", @"HttpHeaderFieldManipulation", @"HttpInvalidRequestLine", @"WebConnectivity"]
 
 @implementation Engine
 
-/** getVersionMK returns the version of Measurement Kit we're using */
-+ (NSString*) versionMK {
-    return [MKVersion versionMK];
-}
-
 /** newUUID4 returns the a new UUID4 for this client  */
 + (NSString*) newUUID4 {
     return OonimkallNewUUID4();
 }
 
-/** startExperimentTask starts the experiment described by the provided settings. */
-+ (id<ExperimentTask>) startExperimentTaskWithSettings:(id<ExperimentSettings>)settings error:(NSError **)error {
-    if ([probeEngineTasks containsObject:settings.taskName]) {
-        return [[OONIProbeEngineTaskAdapter alloc]
-                initWithTask:OonimkallStartTask([settings serialization], error)];
-    }
-    return [[MKExperimentTaskAdapter alloc]
-            initWithTask:[MKAsyncTask start:settings.dictionary]];
+/** startExperimentTaskWithSettings starts the experiment described by the provided settings. */
++ (id<OONIMKTask>) startExperimentTaskWithSettings:(id<OONIMKTaskConfig>)settings error:(NSError **)error{
+    return [[PEMKTask alloc] initWithTask:OonimkallStartTask(settings.serialization, error)];
+}
+
+/** resolveProbeCC returns the probeCC. */
++ (NSString*) resolveProbeCCWithSoftwareName:(NSString*)softwareName
```

Comment:
I am not fully convinced of how errors propagate throughout this function and whether errors may as well be overwritten. I proposed some code that seems to handle the case of errors.